aleph/model/document.py

```diff
@@ -146,6 +153,26 @@ def is_crawler_active(cls, crawler_id):
         return last_run_time > (datetime.utcnow() - timedelta(hours=1))
 
     @classmethod
+    def by_meta(cls, collection_id, meta):
+        q = cls.all()
+        q = q.filter(cls.collection_id == collection_id)
+        if meta.foreign_id:
+            q = q.filter(cls.foreign_id == meta.foreign_id)
+        elif meta.content_hash:
+            q = q.filter(cls.content_hash == meta.content_hash)
+        else:
+            raise ValueError("No unique criterion for document: %s" % meta)
+
+        document = q.first()
+        if document is None:
+            document = Document()
+            document.collection_id = collection_id
+            document.foreign_id = meta.foreign_id
+            document.content_hash = meta.content_hash
+            document.meta = meta
```

Comment:
This changed, I think it is not ok to overwrite the metadata object every time when you can build it from the database metadata field.

https://github.com/alephdata/aleph/blob/master/aleph/ingest/ingestor.py#L61